src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/FixAbstractMethodsStep.cs

```diff
@@ -19,43 +19,80 @@ namespace MonoDroid.Tuner
 	/// <summary>
 	/// NOTE: this step is subclassed so it can be called directly from Xamarin.Android.Build.Tasks
 	/// </summary>
-	public class FixAbstractMethodsStep : BaseStep
+	public class FixAbstractMethodsStep
+#if NET5_LINKER
+	: IMarkHandler
+	{
+		protected LinkContext Context { get; private set; }
+		protected AnnotationStore Annotations => Context?.Annotations;
+
+		public void Initialize (LinkContext context, MarkContext markContext)
+		{
+			Context = context;
+			markContext.RegisterMarkTypeAction (type => ProcessType (type));
+		}
+#else
+	: BaseStep
 	{
 		readonly TypeDefinitionCache cache;
 
 		public FixAbstractMethodsStep (TypeDefinitionCache cache)
 		{
 			this.cache = cache;
 		}
+#endif
 
-		protected override void ProcessAssembly (AssemblyDefinition assembly)
+		bool CheckShouldProcessAssembly (AssemblyDefinition assembly)
 		{
 			if (!Annotations.HasAction (assembly))
 				Annotations.SetAction (assembly, AssemblyAction.Skip);
 
 			if (IsProductOrSdkAssembly (assembly))
-				return;
+				return false;
 
 #if !NET5_LINKER
 			CheckAppDomainUsageUnconditional (assembly, (string msg) => Context.LogMessage (MessageImportance.High, msg));
 #endif
 
+			return assembly.MainModule.HasTypeReference ("Java.Lang.Object");
+		}
+
+		void UpdateAssemblyAction (AssemblyDefinition assembly)
+		{
+			AssemblyAction action = Annotations.HasAction (assembly) ? Annotations.GetAction (assembly) : AssemblyAction.Skip;
```

Comment:
Is Annotations.HasAction needed when you it's already doing same check at https://github.com/xamarin/xamarin-android/pull/5748/files#diff-cad3f2ceb954b4631b88ea1bf3228a3c04085149724a0cf544d6c6d987a46d2bR47 ?

Reply:
Fixed, thanks!